dev_cmds/integration-tests/ts-node-availability.ts

```diff
@@ -0,0 +1,116 @@
+import { execPiped, execToString } from "devcmd";
+import { red } from "kleur/colors";
+import { DOCKER_COMMAND } from "../utils/commands";
+import { NpmPackResult } from "../utils/npm-utils";
+import { singlePackageJsonExampleDir } from "../utils/paths";
+import { TestGroup, TestFunction, installDevcmdCliGlobally, LOCAL_REGISTRY_URL } from "./integration-test-harness";
+
+export function createTsNodeAvailabilityTestGroup(): TestGroup {
+  const setup: TestFunction = async (containerName: string, devcmdCliInfo: NpmPackResult) => {
+    await installDevcmdCliGlobally(containerName, devcmdCliInfo);
+
+    await execPiped({
+      command: DOCKER_COMMAND,
+      args: ["exec", containerName, "sh", "-c", "mkdir /tmp/devcmd_test"],
+    });
+
+    await execPiped({
+      command: DOCKER_COMMAND,
+      args: ["cp", singlePackageJsonExampleDir, `${containerName}:/tmp/devcmd_test`],
+    });
+
+    await execPiped({
+      command: DOCKER_COMMAND,
+      args: ["exec", "--user", "root", containerName, "chown", "-R", "verdaccio", "/tmp/devcmd_test"],
+    });
+
+    await execPiped({
+      command: DOCKER_COMMAND,
+      args: [
+        "exec",
+        containerName,
+        "sh",
+        "-c",
+        ["cd /tmp/devcmd_test/single-package-json", `npm --registry ${LOCAL_REGISTRY_URL} install`].join(" && "),
+      ],
+    });
+
+    return "success";
+  };
+
+  const runExampleCmd: TestFunction = async (containerName: string) => {
+    const exampleCmdCommandLine = `npx devcmd example_cmd`;
+    const { stdout, stderr } = await execToString({
+      command: DOCKER_COMMAND,
+      args: [
+        "exec",
+        containerName,
+        "sh",
+        "-c",
+        ["cd /tmp/devcmd_test/single-package-json", exampleCmdCommandLine].join(" && "),
+      ],
+    });
+
+    if (!stdout.includes("Example command for single-package-json example")) {
+      console.log(red("example_cmd didn't print expected output."));
+      console.log(red("Stderr was:"));
+      console.log(red(stderr));
+
+      return "fail";
+    } else {
+      return "success";
+    }
+  };
```

Comment:
is there a reason why this test case is here (it's duplicated from the single-package-json integration test, right?)

Reply:
I thought of it as a kind of demonstration that the command works in the default setup and fails once you remove ts-node.

However now that I think of it, this is already covered by the other test. Ill remove it :)